python/lsst/ts/watcher/rules/mtm2_tangent_link_temperature.py

```diff
@@ -0,0 +1,164 @@
+    ) -> None:
+
+        # ESS
+        # 106 is the ESS SAL index for the M2 tangent link temperature
+        remote_name_ess = "ESS"
+        remote_info_ess = RemoteInfo(
+            remote_name_ess,
+            106,
+            poll_names=["tel_temperature"],
+        )
+
+        # M2
+        remote_name_m2 = "MTM2"
+        remote_info_m2 = RemoteInfo(
+            remote_name_m2,
+            0,
+            poll_names=["tel_temperature"],
+        )
+
+        super().__init__(
+            config,
+            f"MTM2TangentLinkTemperature.{remote_name_ess}",
+            [remote_info_ess, remote_info_m2],
+            log=log,
+        )
+
+        self._remote_ess: salobj.Remote | None = None
+        self._remote_m2: salobj.Remote | None = None
+
+    @classmethod
+    def get_schema(cls) -> dict[str, typing.Any]:
+        schema_yaml = """
+            $schema: 'http://json-schema.org/draft-07/schema#'
+            description: Configuration for MTM2TangentLinkTemperature rule.
+            type: object
+            properties:
+                buffer:
+                    description: >-
+                        Buffer of the tangent link temperature compared with
+                        the ambient (degree C).
+                    type: number
+                    default: 10.0
+                poll_interval:
+                    description: Time delay between polling updates (second).
+                    type: number
+                    default: 1.0
+
+            required:
+            - buffer
+            - poll_interval
+            additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    def setup(self, model) -> None:
+
+        # 106 is the ESS SAL index for the M2 tangent link temperature
+        self._remote_ess = model.remotes[("ESS", 106)]
+
+        self._remote_m2 = model.remotes[("MTM2", 0)]
+
+    def compute_alarm_severity(self) -> AlarmSeverityReasonType:
+        """Compute and set alarm severity and reason.
+
+        Returns
+        -------
+        None, if no change or unknown, or a tuple of two values:
+
+        severity: `lsst.ts.xml.enums.Watcher.AlarmSeverity`
+            The new alarm severity.
+        reason : `str`
+            Detailed reason for the severity, e.g. a string describing
+            what value is out of range, and what the range is.
+            If ``severity`` is ``NONE`` then this value is ignored (but still
+            required) and the old reason is retained until the alarm is reset
+            to ``nominal`` state.
+
+        Notes
+        -----
+        You may return `NoneNoReason` if the alarm state is ``NONE``.
+        """
+
+        assert self._remote_ess is not None
+        assert self._remote_m2 is not None
+
+        if (not self._remote_ess.tel_temperature.has_data) or (
+            not self._remote_m2.tel_temperature.has_data
+        ):
+            return NoneNoReason
```

Comment:
I think you might want to have a threshold for how long you can stay without data to raise a Warning alarm. If could happen that the ESS is down and we would think everything is ok, when it is not. It might be a good idea to check not only that you have data but also how old the data is. The BaseEssRule does a check like this.

Reply:
I added the timeout in the commit: [ad95027](https://github.com/lsst-ts/ts_watcher/pull/83/commits/ad95027731ce208b62ad4255cfb74e9d0be96976). Thanks!

Reply:
For the checking of data age (or how old it is), I think this would be a generic use case. When I worked on this ticket, I realized the user manual and support for the telemetry data are very limited if someone want to use the polling interface. I also realized there is the cycle dependency between the model and rules in the watcher code.

Therefore, I would suggest we improve the current implementation in watcher to have a common interface to access the telemetry data and its age (extract this part from  BaseEssRule and others. BTW, the user description for BaseEssRule is very limited to understand how to use it).

If you agree, I could write a ticket to provide a common user interface of telemetry to assign to myself to improve the watcher. I can improve the developer guide that how to use the polling in watcher as well.

Please let me know your idea. Thanks!